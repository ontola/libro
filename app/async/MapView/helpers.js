import * as fa from 'fontawesome';
import Circle from 'ol/geom/Circle';
import ImageState from 'ol/ImageState';
import { toContext } from 'ol/render';
import CircleStyle from 'ol/style/Circle';
import Fill from 'ol/style/Fill';
import Icon from 'ol/style/Icon';
import { get as getIconImage } from 'ol/style/IconImage';
import Stroke from 'ol/style/Stroke';
import Style from 'ol/style/Style';
import Text from 'ol/style/Text';
import { useLayoutEffect, useMemo } from 'react';

import {
  isFontAwesomeIRI,
  normalizeFontAwesomeIRI,
} from '../../helpers/iris';

const ANCHOR_Y_BOTTOM = 1;
const ANCHOR_X_CENTER = 0.5;
const CIRCLE_RADIUS = 12;
const CIRCLE_SIZE = 13;
const ICON_X = CIRCLE_SIZE;
const ICON_Y = CIRCLE_SIZE;
const IMG_SIZE = 26;

const iconCache = {};

const getImage = (image) => {
  const img = getIconImage(null, image, IMG_SIZE, true, ImageState.IDLE, null);
  img.load();

  return img;
};

const generateImageStyle = (image, highlight, count, theme) => {
  const canvas = document.createElement('canvas');
  const canvasCtx = canvas.getContext('2d');

  if (highlight) {
    canvasCtx.globalAlpha = 0.6;
  }
  const img = getImage(image);
  canvasCtx.drawImage(img.getImage(), 0, 0, IMG_SIZE, IMG_SIZE);
  canvasCtx.globalAlpha = 1;

  const numberText = count > 1 && (
    new Text({
      fill: new Fill({ color: 'white' }),
      font: `bold 16px ${theme.typography.h1.fontFamily}`,
      offsetX: CIRCLE_RADIUS,
      text: count.toString(),
    })
  );

  return {
    image: new Icon({
      anchor: [ANCHOR_X_CENTER, ANCHOR_Y_BOTTOM],
      img: canvas,
      imgSize: [IMG_SIZE, IMG_SIZE],
    }),
    text: numberText,
  };
};

const generateFontAwesomeStyle = (image, highlight, count, theme) => {
  const canvas = document.createElement('canvas');
  const canvasCtx = canvas.getContext('2d');

  const circleBackground = new Fill({ color: highlight ? '#92a1b5' : '#475668' });
  const circleStroke = new Stroke({
    color: 'white',
    width: 2,
  });
  const circleStyle = new Style({
    fill: circleBackground,
    image: new CircleStyle({
      fill: circleBackground,
      radius: 10,
      stroke: circleStroke,
    }),
    stroke: circleStroke,
  });

  const circle = new Circle([CIRCLE_SIZE, CIRCLE_SIZE], CIRCLE_RADIUS);

  const vectorContext = toContext(canvasCtx, {
    pixelRatio: 1,
    size: [100, 100],
  });
  vectorContext.setStyle(circleStyle);
  vectorContext.drawGeometry(circle);

  let renderText, renderFont;
  if (count > 1) {
    renderText = count;
    renderFont = `bold 16px ${theme.typography.h1.fontFamily}`;
  } else {
    renderText = fa(normalizeFontAwesomeIRI(image));
    renderFont = 'normal 18px FontAwesome';
  }
  /* eslint-disable no-param-reassign */
  canvasCtx.fillStyle = 'white';
  canvasCtx.font = renderFont;
  canvasCtx.textAlign = 'center';
  canvasCtx.textBaseline = 'middle';
  /* eslint-enable no-param-reassign */
  canvasCtx.fillText(renderText, ICON_X, ICON_Y);

  return {
    image: new Icon({
      anchor: [ANCHOR_X_CENTER, ANCHOR_Y_BOTTOM],
      img: canvas,
      imgSize: [IMG_SIZE, IMG_SIZE],
    }),
  };
};

const generateIconStyle = (image, highlight = false, count, theme) => {
  let style;
  if (isFontAwesomeIRI(image)) {
    style = generateFontAwesomeStyle(image, highlight, count, theme);
  } else {
    style = generateImageStyle(image, highlight, count, theme);
  }

  return new Style(style);
};

const featureCount = (feature) => (
  feature?.get('features')?.length || 0
);

const getStyle = (image, highlight, theme) => (feature) => {
  const count = featureCount(feature);
  const cacheKey = [image, highlight, count].filter(Boolean).join('-');

  if (!iconCache[cacheKey]) {
    const icon = generateIconStyle(image, highlight, count, theme);

    iconCache[cacheKey] = icon;
  }

  return iconCache[cacheKey];
};

export const useImageStyles = (images, theme) => {
  const normalizedImages = useMemo(() => images.map((image) => image?.value || image), []);
  useLayoutEffect(() => {
    normalizedImages.forEach((image) => {
      if (image && !isFontAwesomeIRI(image)) {
        getImage(image);
      }
    });
  }, normalizedImages);

  const hoverStyles = useMemo(() => {
    const result = {};
    normalizedImages.forEach((image) => {
      result[image] = getStyle(image, false, theme);
    });

    return result;
  }, normalizedImages);
  const styles = useMemo(() => {
    const result = {};
    normalizedImages.forEach((image) => {
      result[image] = getStyle(image, true, theme);
    });

    return result;
  }, normalizedImages);

  return {
    hoverStyles,
    styles,
  };
};
