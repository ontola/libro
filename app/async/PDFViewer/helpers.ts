import React from "react";
import { History } from "history";

/** Mapping for consistent component Ids, used by reactivesearch */
export const ids = {
  searchbox: "zoekterm",
  organisaties: "organisaties",
  daterange: "datums",
  type: "type",
  tag: "thema",
  location: "locatie",
};

export function usePersistedState<T>(key: string, initial: T):
  [T, React.Dispatch<React.SetStateAction<T>>] {
  const [value, setValue] = React.useState<T>(() => {
    const storageValue = sessionStorage.getItem(key);
    if (storageValue) {
      return JSON.parse(storageValue) || initial;
    }

    return initial;
  });

  const setPersistedValue = (next: T | React.SetStateAction<T>): void => {
    sessionStorage.setItem(key, JSON.stringify(next));
    setValue(next);
  };
  return [value, setPersistedValue];
}

export function myPersistedState<T>(key: string, initial: T): T {
  const storageValue = sessionStorage.getItem(key);
  if (storageValue === null) {
    return initial;
  }
  return JSON.parse(storageValue);
}

export const getTopicsApiURL = (relUrl: string): URL => {
  const url = new URL(window.location.origin);
  url.pathname = `/topics_api/dev${relUrl}`;

  return url
};

export const getParams = (history: History) => {
  const searchObject = history.location.search;
  const hasParams = searchObject !== "";
  const params = new URLSearchParams(searchObject);

  const currentResourceBase = params.get("showResource");
  let currentResource = null;
  if (currentResourceBase !== null) {
    currentResource = decodeURIComponent(currentResourceBase);
  }

  let currentSearchTerm = params.get(ids.searchbox);
  if (currentSearchTerm) {
    currentSearchTerm = currentSearchTerm.substr(1, currentSearchTerm.length - 2);
  }

  let documentID = null;
  if (currentResource !== null) {
    documentID = currentResource.split("/")[3]
  }

  return {
    currentResource,
    currentSearchTerm,
    hasParams,
    documentID
  };
};
