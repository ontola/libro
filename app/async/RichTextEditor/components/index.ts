export * from './EditableWithPlugins';
export * from './ElementButton';
export * from './InputDialog';
export * from './ListButton';
export * from './MarkButton';
export * from './RichTextEditor';
export * from './RichTextEditorMd';
export * from './ToggleButton';
export * from './ToggleButtonWithInputDialog';
