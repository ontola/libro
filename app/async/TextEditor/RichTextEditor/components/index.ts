export * from './EditableWithPlugins';
export * from './RichTextEditor';
export * from './RichTextEditorMd';
