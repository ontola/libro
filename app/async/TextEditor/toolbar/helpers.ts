import { makeStyles } from '@material-ui/core';
import { isNode, NamedNode, Node } from '@ontologies/core';
import { LinkReduxLRSType, useLRS } from 'link-redux';
import { EventHandler } from 'react';
import { useSlate } from 'slate-react';

export const useMenuItemStyles = makeStyles(({
  button: {
    border: 'none',
    height: 48,
  },
  root: {
    backgroundColor: 'transparent',
  },
}));

export function normalizeAction(
  lrs: LinkReduxLRSType,
  action: Node | EventHandler<any>,
  opts: any,
): EventHandler<any> {
  if (isNode(action)) {
    return (e: Event) => {
      e.preventDefault();

      return lrs.exec(action as NamedNode, opts);
    };
  }

  return action;
}

export const useNormalizeAction = (action: EventHandler<any>, item: any) => {
  const lrs = useLRS();
  const textEditor = useSlate();
  const actionOpts = { item, textEditor };

  return normalizeAction(lrs, action, actionOpts);
};
