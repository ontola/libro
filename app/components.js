export { default as ActionsBar } from './components/ActionsBar/index';
// export { default as Attachment } from './components/Attachment/index';
export { default as Attachment } from './components/Attachment/index';
export { default as BackButton } from './components/BackButton/index';
export { default as BottomBar } from './components/BottomBar/index';
export { default as BottomBarButton } from './components/BottomBarButton/index';
export { default as Button } from './components/Button/index';
export { default as BlurButton } from './components/BlurButton/index';
export { default as Card } from './components/Card/index';
export { default as CardActions } from './components/Card/CardActions';
export { default as CardAppendix } from './components/Card/CardAppendix';
export { default as CardButton } from './components/Card/CardButton';
export { default as CardContent } from './components/Card/CardContent';
export { default as CardDivider } from './components/Card/CardDivider';
export { default as CardFixed } from './components/Card/CardFixed';
export { default as CardLink } from './components/Card/CardLink';
export { default as CardMain } from './components/Card/CardMain';
export { default as CardMenuFloater } from './components/Card/CardMenuFloater';
export { default as CardRow } from './components/Card/CardRow';
export { default as CardMicroRow } from './components/Card/CardMicroRow';
export { default as Carousel } from './components/Grid/Carousel';
// export { default as ChronoFeed } from './components/ChronoFeed/index';
// export { default as ChronoItem } from './components/ChronoItem/index';
export { default as Collapsible } from './components/Collapsible/index';
export { default as CollapseText } from './components/CollapseText/index';
export { default as Column } from './components/Column/index';
export { default as Columns } from './components/Columns/index';
// export { default as CompareVotesBar } from './components/CompareVotesBar/index';
export { default as Container } from './components/Container/index';
export { default as CoverImage } from './components/CoverImage/index';
// export { default as CountBubble } from './components/CountBubble/index';
export { default as Cover } from './components/Cover/index';
export { default as Detail } from './components/Detail/index';
export { default as DetailText } from './components/Detail/text';
export { default as DetailImage } from './components/Detail/image';
export { default as DetailDate } from './components/DetailDate/index';
// export { default as DetailDuration } from './components/DetailDuration/index';
export { default as DetailsBar } from './components/DetailsBar/index';
export { default as DetailStatus } from './components/DetailStatus/index';
export { default as DetailType } from './components/DetailType/index';
export { default as DetailVotedFor } from './components/DetailVotedFor/index';
// export { default as DonutChart } from './components/DonutChart/index';
export { default as Drawer } from './components/Drawer/index';
export { default as Dropdown } from './components/Dropdown/index';
export { default as DropdownLink } from './components/Dropdown/DropdownLink';
export { default as Error } from './components/Error/index';
// export { default as EventItemShow } from './components/EventItemShow/index';
export {
  Form,
  FormFooter,
  FormSection,
} from './components/Form/index';
export { default as FormField } from './components/FormField/index';
export { default as Grid } from './components/Grid/index';
export { default as GeneratedIcon } from './components/GeneratedIcon/index';
export { default as Header } from './components/Header/index';
export { default as Heading } from './components/Heading/index';
export { default as HiddenFormField } from './components/HiddenFormField/index';
export { default as hoverBox } from './components/HoverBox/index';
export { default as Image } from './components/Image/index';
export {
  Input,
  FileInput,
} from './components/Input/index';
export { default as LabelValue } from './components/LabelValue/index';
export { default as LabelValueBar } from './components/LabelValueBar/index';
export { default as LDLink } from './components/LDLink/index';
// export { default as LinkedDetailDate } from './components/LinkedDetailDate/index';
export { default as HoverPopup } from './components/LinkedPopup/HoverPopup';
export { default as LinkedPopup } from './components/LinkedPopup/index';
export { default as LinkCard } from './components/LinkCard/index';
export { default as LinkDuo } from './components/LinkDuo/index';
export { default as LinkList } from './components/LinkList/index';
export { default as List } from './components/List/index';
export {
  default as Loading,
  LoadingCard,
  LoadingCardFixed,
  LoadingMicroRow,
  LoadingDetail,
  LoadingPage,
  LoadingParent,
  LoadingRow,
  LoadingSideBarLink,
} from './components/Loading/index';
export { default as Markdown } from './components/Markdown/index';
export { default as MarkdownFixedPreview } from './components/MarkdownFixedPreview/index';
export { default as NavBarContent } from './components/NavBarContent/index';
export { default as Notification } from './components/Notification/index';
export { default as Omniform } from './components/Omniform/index';
export { default as OmniformFields } from './components/Omniform/OmniformFields';
export { default as OmniformPreview } from './components/Omniform/OmniformPreview';
export { default as OrgSwitcher } from './components/OrgSwitcher/index';
export { default as PageHeader } from './components/PageHeader/index';
export { default as ProfileCard } from './components/ProfileCard/index';
export { default as ProfileListItem } from './components/ProfileListItem/index';
export { default as Progress } from './components/Progress/index';
export { default as ProgressBar } from './components/ProgressBar/index';
// export { default as SearchResultItem } from './components/SearchResultItem/index';
export { default as Resource } from './components/Resource/index';
export { default as SideBar } from './components/SideBar/index';
export { default as SideBarLink } from './components/SideBarLink/index';
export { default as SideBarCollapsible } from './components/SideBarCollapsible/index';
export {
  SignInFormCard,
  SignInFormCardRow,
} from './components/SignInForm/index';
export { default as SkipNavigation } from './components/SkipNavigation/index';
export { default as Spinner } from './components/Spinner/index';
export { default as TabBar } from './components/TabBar/index';
export { default as Tab } from './components/TabBar/Tab';
export { default as Tag } from './components/Tag/index';
export { default as TextEditor } from './components/TextEditor/index';
// export { default as VideoPlayer } from './components/VideoPlayer/index';
// export { default as VoteChart } from './components/VoteChart/index';
// export { default as VoteData } from './components/VoteData/index';
export { default as WidgetTopology } from './components/Widget/WidgetTopology';
