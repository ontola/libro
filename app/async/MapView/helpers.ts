import { Theme } from '@material-ui/core';
import fa from 'fontawesome';
import { Feature } from 'ol';
import Circle from 'ol/geom/Circle';
import { toContext } from 'ol/render';
import CircleStyle from 'ol/style/Circle';
import Fill from 'ol/style/Fill';
import Icon from 'ol/style/Icon';
import Stroke from 'ol/style/Stroke';
import Style from 'ol/style/Style';

import { isFontAwesomeIRI, normalizeFontAwesomeIRI } from '../../helpers/iris';

const ANCHOR_Y_BOTTOM = 1;
const ANCHOR_X_CENTER = 0.5;
const CIRCLE_RADIUS = 12;
const CIRCLE_SIZE = 13;
const ICON_X = CIRCLE_SIZE;
const ICON_Y = CIRCLE_SIZE;
const IMG_SIZE = 26;

const iconCache: { [k: string]: Style } = {};

const drawFontAwesomeIcon = (
  canvasCtx: CanvasRenderingContext2D,
  text: string,
  highlight: boolean,
  count: number,
  theme: any,
) => {
  const circleBackground = new Fill({ color: highlight ? '#92a1b5' : '#475668' });
  const circleStroke = new Stroke({
    color: 'white',
    width: 2,
  });
  const circleStyle = new Style({
    fill: circleBackground,
    image: new CircleStyle({
      fill: circleBackground,
      radius: 10,
      stroke: circleStroke,
    }),
    stroke: circleStroke,
  });

  const circle = new Circle([CIRCLE_SIZE, CIRCLE_SIZE], CIRCLE_RADIUS);

  const vectorContext = toContext(canvasCtx, {
    pixelRatio: 1,
    size: [100, 100],
  });
  vectorContext.setStyle(circleStyle);
  vectorContext.drawGeometry(circle);

  let renderText;
  let renderFont;
  if (count > 1) {
    renderText = count.toString();
    renderFont = `bold 16px ${theme.typography.h1.fontFamily}`;
  } else {
    renderText = text;
    renderFont = 'normal 18px FontAwesome';
  }
  /* eslint-disable no-param-reassign */
  canvasCtx.fillStyle = 'white';
  canvasCtx.font = renderFont;
  canvasCtx.textAlign = 'center';
  canvasCtx.textBaseline = 'middle';
  /* eslint-enable no-param-reassign */
  canvasCtx.fillText(renderText, ICON_X, ICON_Y);
};

const generateIconStyle = (image: string, highlight = false, count: number, theme: any) => {
  const canvas = document.createElement('canvas');
  const canvasCtx = canvas.getContext('2d');

  if (canvasCtx && isFontAwesomeIRI(image)) {
    drawFontAwesomeIcon(canvasCtx, fa(normalizeFontAwesomeIRI(image)), highlight, count, theme);
  }

  return new Style({
    image: new Icon({
      anchor: [ANCHOR_X_CENTER, ANCHOR_Y_BOTTOM],
      img: canvas,
      imgSize: [IMG_SIZE, IMG_SIZE],
    }),
  });
};

const featureCount = (feature: Feature) => (
  feature?.get('features')?.length || 0
);

const getStyle = (image: string, highlight: boolean, theme: any) => (feature: Feature) => {
  const count = featureCount(feature);
  const cacheKey = [image, highlight, count].filter(Boolean).join('-');

  if (!iconCache[cacheKey]) {
    const icon = generateIconStyle(image, highlight, count, theme);

    iconCache[cacheKey] = icon;
  }

  return iconCache[cacheKey];
};

export const getStyles = (image: string, theme: Theme): {
  hoverStyle: (feature: Feature) => Style,
  style: (feature: Feature) => Style,
} => {
  const style = getStyle(image, false, theme);
  const hoverStyle = getStyle(image, true, theme);

  return {
    hoverStyle,
    style,
  };
};
