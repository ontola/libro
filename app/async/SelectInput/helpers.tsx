import { FilterOptionsState } from '@material-ui/lab';
import { SomeTerm, isLiteral } from '@ontologies/core';
import * as rdfx from '@ontologies/rdf';
import * as schema from '@ontologies/schema';
import { SomeNode } from 'link-lib';
import { Resource, useLRS } from 'link-redux';
import React from 'react';
import { defineMessages } from 'react-intl';

import { entityIsLoaded } from '../../helpers/data';
import { isResource } from '../../helpers/types';
import ontola from '../../ontology/ontola';

const messages = defineMessages({
  noMatchingItems: {
    defaultMessage: 'No matching items',
    id: 'https://app.argu.co/i18n/forms/select/noMatchingItems',
  },
  typeToSearch: {
    defaultMessage: 'Type to start searching',
    id: 'https://app.argu.co/i18n/forms/select/typeToSearch',
  },
});

const stripDiacritics = (str: string) => (
  typeof str.normalize !== 'undefined'
    ? str.normalize('NFD').replace(/[\u0300-\u036f]/g, '')
    : str
);

export const filterOptions = (
  options: SomeTerm[],
  { inputValue, getOptionLabel }: FilterOptionsState<SomeTerm>,
): SomeTerm[] => {
  const input = stripDiacritics(inputValue.toLowerCase());

  return options.filter((option) => {
    const candidate = stripDiacritics(getOptionLabel(option).toLowerCase());

    return candidate.indexOf(input) > -1;
  });
};

export const useItemToString = (): ((i: SomeTerm | undefined | null) => string) => {
  const lrs = useLRS();

  return React.useCallback((item: SomeTerm | undefined | null) => {
    if (typeof item === 'undefined' || item === null) {
      return '';
    }

    if (!isResource(item)) {
      return item.value;
    }

    if (!entityIsLoaded(lrs, item)) {
      return 'Loading';
    }

    const itemClass = lrs.getResourceProperty<SomeNode>(item, rdfx.type);
    const itemClassDisplayProp = lrs.getResourceProperty<SomeNode>(
      itemClass,
      ontola['forms/inputs/select/displayProp'],
    );
    const classDisplayProp = itemClassDisplayProp ?? schema.name;
    const label = lrs.getResourceProperty(item, [classDisplayProp, schema.name]);

    return label?.value ?? item.value;
  }, [lrs]);
};

export function emptyMessage(
  fmt: (args: any) => string,
  searchable: boolean,
  currentValue: string,
): string {
  if (searchable) {
    if (currentValue && currentValue.length > 0) {
      return fmt(messages.noMatchingItems);
    }

    return fmt(messages.typeToSearch);
  }

  return fmt(messages.noMatchingItems);
}

export const renderOption = (item: SomeTerm): JSX.Element => {
  if (isLiteral(item.termType)) {
    return (
      <option
        className="Field__list-element"
        key={item.value}
        value={item.value}
      >
        {item.value}
      </option>
    );
  }

  return (
    <Resource
      element="div"
      key={item.value}
      subject={item}
    />
  );
};
