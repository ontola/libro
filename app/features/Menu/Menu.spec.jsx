import { Literal, NamedNode } from 'rdflib';
import { create } from 'react-test-renderer';

import { NS } from '../../helpers/LinkedRenderStore';
import { createContext } from '../../../tests/specHelper';

import components from './Menu';

const resource = new NamedNode('https://argu.dev/o/1/menus/navigations#forums');

const resources = {
  [resource]: {
    [NS.rdf('type')]: NS.argu('MenuSection'),
    [NS.schema('name')]: new Literal('Forum'),
    [NS.schema('isPartOf')]: new NamedNode('https://argu.dev/o/1/menus/navigations'),
    [NS.argu('menuItems')]: [
      new NamedNode('https://argu.dev/o/1/menus/navigations#forums.motions'),
      new NamedNode('https://argu.dev/o/1/menus/navigations#forums.questions'),
      new NamedNode('https://argu.dev/o/1/menus/navigations#forums.settings'),
    ],
  },
  [new NamedNode('https://argu.dev/o/1/menus/navigations#forums.motions')]: {
    [NS.rdf('type')]: NS.argu('MenuItem'),
    [NS.schema('name')]: new Literal('IdeeÃ«n'),
    [NS.schema('url')]: new Literal('https://argu.dev/f/utrecht/motions'),
    [NS.schema('isPartOf')]: new NamedNode('https://argu.dev/o/1/menus/navigations#forums'),
    [NS.schema('image')]: new NamedNode('http://fontawesome.io/icon/lightbulb-o'),
  },
  [new NamedNode('https://argu.dev/o/1/menus/navigations#forums.questions')]: {
    [NS.rdf('type')]: NS.argu('MenuItem'),
    [NS.schema('name')]: new Literal('Vraagstukken'),
    [NS.schema('url')]: new Literal('https://argu.dev/f/utrecht/questions'),
    [NS.schema('isPartOf')]: new NamedNode('https://argu.dev/o/1/menus/navigations#forums'),
    [NS.schema('image')]: new NamedNode('http://fontawesome.io/icon/question'),
  },
  [new NamedNode('https://argu.dev/o/1/menus/navigations#forums.settings')]: {
    [NS.rdf('type')]: NS.argu('MenuItem'),
    [NS.schema('name')]: new Literal('Instellingen'),
    [NS.schema('url')]: new Literal('https://argu.dev/utrecht/settings'),
    [NS.schema('isPartOf')]: new NamedNode('https://argu.dev/o/1/menus/navigations#forums'),
    [NS.schema('image')]: new NamedNode('http://fontawesome.io/icon/gear'),
  },
};

describeView('Menu', components, resources, resource, () => {
  as(NS.argu('sidebar'), () => {
    it('renders a menu', () => {
      expect(create(createContext())).toMatchSnapshot();
    });
  });
});
