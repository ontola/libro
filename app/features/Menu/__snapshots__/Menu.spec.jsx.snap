// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu https://argu.co/ns/core#sidebar renders a menu 1`] = `
<div
  data-test="MenuSection-menu-section"
>
  <div
    className="MenuSectionLabel"
    data-test="MenuSection-MenuSectionLabel"
  >
    <div
      className="MenuSectionLabel__bar"
    />
    <div
      className="MenuSectionLabel__text"
    >
      Forum
    </div>
    <div
      className="MenuSectionLabel__bar"
    />
  </div>
  <div
    className="SideBarLink"
  >
    <a
      aria-current={null}
      className="SideBarLink__link"
      href="/f/utrecht/motions"
      onClick={[Function]}
    >
      <div
        className="SideBarLink__icon"
      >
        <span
          aria-hidden={true}
          className="fa fa-lightbulb-o"
        />
      </div>
      IdeeÃ«n
    </a>
  </div>
  <div
    className="SideBarLink"
  >
    <a
      aria-current={null}
      className="SideBarLink__link"
      href="/f/utrecht/questions"
      onClick={[Function]}
    >
      <div
        className="SideBarLink__icon"
      >
        <span
          aria-hidden={true}
          className="fa fa-question"
        />
      </div>
      Vraagstukken
    </a>
  </div>
  <div
    className="SideBarLink"
  >
    <a
      aria-current={null}
      className="SideBarLink__link"
      href="/utrecht/settings"
      onClick={[Function]}
    >
      <div
        className="SideBarLink__icon"
      >
        <span
          aria-hidden={true}
          className="fa fa-gear"
        />
      </div>
      Instellingen
    </a>
  </div>
</div>
`;
